django/db/models/expressions.py

```diff
@@ -424,6 +420,17 @@ class Expression(BaseExpression, Combinable):
 
 class CombinedExpression(SQLiteNumericMixin, Expression):
 
+    _connector_output_fields = {
+        (connector, lhs, rhs): output_field
+        for connector in (Combinable.ADD, Combinable.SUB, Combinable.MUL, Combinable.DIV)
+        for lhs, rhs, output_field in (
+            (fields.IntegerField, fields.DecimalField, fields.DecimalField),
+            (fields.DecimalField, fields.IntegerField, fields.DecimalField),
+            (fields.IntegerField, fields.FloatField, fields.FloatField),
+            (fields.FloatField, fields.IntegerField, fields.FloatField),
```

Comment:
Do we want to handle other cases here such as `BigIntegerField + IntegerField → BigIntegerField`? There are many more possible combinations...

Also, when `_connector_output_fields` is used, `isinstance()` isn't used so `BigIntegerField + FloatField` will raise. This is actually fine if we want to handle cases such as using the larger size integer field as above, but would require many more entries here.

Reply:
I think we should keep it simple, I would even consider removing `_connector_output_fields` :thinking:

Reply:
I'd remove it and defer all dispatching efforts to ticket-26355 (#6395) but this is the strict minimum to get the tests passing IIRC.

Reply:
@pope1ni, @felixxm just pushed a `!fixup` that addresses Nick's concern while making the `annotations` tests pass.